padertorch/data/fragmenter.py

```diff
@@ -15,110 +16,199 @@
+        >>> ex = {'x': np.arange(63999), 'y': np.arange(63999)}
+        >>> for entry in segmenter(ex):
+        ...     print(entry['x'][[0, -1]])
+        [    0 31999]
+        [16000 47999]
+
+    Args:
+        length: The length of the segments in samples. If set to `-1`,
+            the original example is returned in a list of length one.
+        shift: shift between segments, defaults to length
+        include_keys: The keys in the passed example dict to segment. They all
+            must have the same size along their specified `axis`, if keys is
+            None the segmentation is applied to all `numpy.arrays`.
+            If a key points to a dictionary the segmentation is applied to all
+            values of this dictionary
+        exclude_keys: This option allows to specify specific keys not to
+            segment. This might be usefull if `include_keys` is `None` or
+            one of the included keys points to a dictionary.
+        copy_keys: If `None` all not values which are not segmented are
+            copied. Otherwise only the specified keys are added to the new
+            dictionary with the segmented signals.
+        axis: axis over which to segment
+        anchor_mode: anchor mode used in `get_anchor` to calculate the anchor
+            from which the segmentation boundaries are calculated.
+        rng: random number generator (`numpy.random`)
     """
-    def __init__(
-            self, fragment_steps, fragment_lengths=None, axis=-1,
-            squeeze=False, drop_last=False, copy_keys=None
-    ):
-        self.fragment_steps = fragment_steps
-        self.fragment_lengths = fragment_lengths \
-            if fragment_lengths is not None else fragment_steps
-        self.axis = axis
-        self.squeeze = squeeze
-        self.drop_last = drop_last
-        self.copy_keys = copy_keys
-
-    def __call__(self, example, random_onset=False):
-        copies = flatten(
-            {key: example[key] for key in self.copy_keys}
-            if self.copy_keys is not None else example
-        )
 
-        if random_onset:
-            start = np.random.rand()
-            for fragment_step in self.fragment_steps.values():
-                start = int(int(start*fragment_step) / fragment_step)
+    def __init__(self, length: int = -1,shift: int = None,
+                 include_keys: Union[str, list, tuple] = None,
+                 exclude_keys: Union[str, list, tuple] = None,
+                 copy_keys: Union[str, list, tuple] = None,
+                 axis: Union[int, list, tuple, dict] = -1,
+                 anchor_mode: str = 'left', rng=np.random):
+
+        self.include = None if include_keys is None else to_list(include_keys)
+        self.exclude = [] if exclude_keys is None else to_list(exclude_keys)
+        self.length = length
+        if isinstance(axis, (dict, int)):
+            self. axis = axis
+        elif isinstance(axis, (tuple, list)):
+            self.axis = to_list(axis)
```

Comment:
`self.axis = to_list(axis)` -> `self.axis = list(axis)`
Not sure, why you make this operation.

Reply:
removes the need to ask whether it might be a tuple in the rest of the code